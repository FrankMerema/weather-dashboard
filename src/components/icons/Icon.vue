<template>
  <svg class="ws-icon" :class="{ chameleon }">
    <use :xlink:href="iconHref" />
  </svg>
</template>

<script>
export default {
  props: {
    chameleon: {
      default: false,
      required: false,
      type: Boolean
    },
    icon: {
      required: true,
      type: String
    }
  },
  computed: {
    iconHref() {
      return `#ws-${this.icon}`;
    }
  }
};
</script>

<style lang="scss" scoped>
svg {
  &.ws-icon {
    display: inline-flex;
    max-height: 100%;
    width: 100%;
  }

  &.chameleon {
    fill: currentColor;
  }
}
</style>
