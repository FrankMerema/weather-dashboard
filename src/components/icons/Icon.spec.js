import { mount } from '@vue/test-utils';
import Icon from './Icon';

describe('Icon', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Icon, { propsData: { icon: 'test' } });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('check if svg icon gets prefixed', () => {
    const wrapper = mount(Icon, { propsData: { icon: 'test' } });

    expect(wrapper.find('svg use').attributes('href')).toEqual('#ws-test');
  });
});

describe('Icon snapshot', () => {
  test('render logo icon correctly', () => {
    const wrapper = mount(Icon, {
      propsData: {
        icon: 'logo'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
