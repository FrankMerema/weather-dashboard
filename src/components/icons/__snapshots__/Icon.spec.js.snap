// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon snapshot render logo icon correctly 1`] = `
<svg
  class="ws-icon"
>
  <use
    xlink:href="#ws-logo"
  />
</svg>
`;
